### PR TITLE
Review Dockerfile and RPM files for building odh-modelmesh-serving-controller in onflux

### PR DESCRIPTION
**Pull request by MohammadiIram**
Resolves #https://issues.redhat.com/browse/RHOAIENG-14146

### CODE REVIEW

**Comment by spolti** [comment]
Hi, it is still a draft, is this PR ready for review?

**Comment by MohammadiIram** [comment]
> Hi, it is still a draft, is this PR ready for review?

Hello, Please review the PR

**Comment by MohammadiIram** [comment]
> Hi, it is still a draft, is this PR ready for review?

Hello, Please review the PR

File: Dockerfile.konflux [hunk 1]
```diff
@@ -0,0 +1,64 @@
+ARG DEV_IMAGE=registry.redhat.io/ubi9/go-toolset@sha256:fd41c001abc243076cc28b63c409ae6d9cbcad401c8124fb67d20fe57a2aa63a
```
Comment (spolti) [inline Dockerfile.konflux:1]: if we are going to have a Containerfile specific to konflux, we can drop this change.

Comment (spolti) [inline Dockerfile.konflux:1]: can't we call this file as Containerfile?
It is better than dockerfile, and it is aligned with Podman.

File: Dockerfile.konflux [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+FROM registry.redhat.io/ubi9/go-toolset@sha256:fd41c001abc243076cc28b63c409ae6d9cbcad401c8124fb67d20fe57a2aa63a AS build
```
Comment (spolti) [inline Dockerfile.konflux:3]: Please add a comment referring to the tag this image refers to.

File: Dockerfile.konflux [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+RUN GOOS=${TARGETOS:-linux} \
+    GOARCH=${TARGETARCH:-amd64} \
+    CGO_ENABLED=0 \
+    GO111MODULE=on \
+    go build -a -o /workspace/manager main.go
+
+###############################################################################
+# Stage 2: Copy build assets to create the smallest final runtime image
+###############################################################################
+FROM registry.redhat.io/ubi8/ubi-minimal@sha256:7583ca0ea52001562bd81a961da3f75222209e6192e4e413ee226cff97dbd48c AS runtime
```
Comment (spolti) [inline Dockerfile.konflux:36]: same here, comment with the tag.

File: Dockerfile.konflux [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      summary="odh-modelmesh-serving-controller" \
+      maintainer="['managed-open-data-hub@redhat.com']" \
+      io.openshift.expose-services="" \
+      io.k8s.display-name="modelmesh-serving-controller" \
+      io.k8s.description="modelmesh-serving-controller" \
+      com.redhat.license_terms="https://www.redhat.com/licenses/Red_Hat_Standard_EULA_20191108.pdf"
+
+
+## Install additional packages
+RUN microdnf install -y shadow-utils &&\
```
Comment (spolti) [inline Dockerfile.konflux:54]: why are you adding this if no user is being added?

File: ubi.repo [hunk 5]
```diff
@@ -0,0 +1,71 @@
+[ubi-8-baseos-rpms]
```
Comment (spolti) [inline ubi.repo:1]: as a minimal number of packages are being added, do we need to define all these repos?

File: Dockerfile.konflux [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    microdnf clean all
+
+USER ${USER}
+
+WORKDIR /
+COPY --from=0 /workspace/manager .
```
[the remark this answers is not in the document]
Comment (MohammadiIram) [reply Dockerfile.konflux:60]: done

File: Dockerfile.konflux (hunk 1 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:1]: YES

File: Dockerfile.konflux (hunk 2 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:3]: SURE

File: Dockerfile.konflux (hunk 1 above)
Comment (MohammadiIram) [reply Dockerfile.konflux]: We trying to keep standardized naming across all components as Dockerfile.konflux.

File: Dockerfile.konflux (hunk 4 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:53]: I have not added it, referred from the midstream dockerfile

File: ubi.repo (hunk 5 above)
Comment (MohammadiIram) [reply ubi.repo:1]: yes, this is an autogenerated sowe don't want to change anything

File: Dockerfile.konflux [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+RUN GOOS=${TARGETOS:-linux} \
+    GOARCH=${TARGETARCH:-amd64} \
+    CGO_ENABLED=0 \
+    GO111MODULE=on \
+    go build -a -o /workspace/manager main.go
+
+###############################################################################
+# Stage 2: Copy build assets to create the smallest final runtime image
+###############################################################################
+#ubi-minimal:latest
```
Comment (spolti) [inline Dockerfile.konflux:34]: which tag is latest? 8.8, 8.9?

File: Dockerfile.konflux (hunk 4 above)
Comment (spolti) [reply Dockerfile.konflux:53]: we are not adding users, it can be removed..

**Pull request merged**